src/darwin/Framework/CHIP/MTRDevice_Concrete.mm

```diff
@@ -3566,10 +3790,15 @@ - (void)_pruneEndpointsIn:(MTRDeviceDataValueDictionary)previousPartsListValue
         [self _removeClusters:clusterPathsToRemove doRemoveFromDataStore:NO];
         [self._concreteController.controllerDataStore clearStoredClusterDataForNodeID:self.nodeID endpointID:endpoint];
 
+        mtr_weakify(self);
         [_deviceController asyncDispatchToMatterQueue:^{
-            std::lock_guard lock(self->_lock);
-            if (self->_currentSubscriptionCallback) {
-                self->_currentSubscriptionCallback->ClearCachedAttributeState(static_cast<EndpointId>(endpoint.unsignedLongLongValue));
+            mtr_strongify(self);
+            VerifyOrReturn(self);
+
+            @synchronized(self.matterCPPObjectsHolder) {
```

Comment:
Again, why do we need to synchronize if all the accesses are on the Matter queue?  I'll stop commenting on this.

Reply:
See https://github.com/project-chip/connectedhomeip/issues/37420